packages/ckeditor5-emoji/src/emojirepository.ts

```diff
@@ -264,6 +291,26 @@ async function loadEmojiDatabase( emojiDatabaseUrl: string ): Promise<Array<Emoj
 	return result;
 }
 
+/**
+ * Checks the supported emoji version by the OS, by sampling some representatives from different emoji releases.
+ */
+function getEmojiSupportedVersionByOs() {
+	return Object.entries( EMOJI_SUPPORT_LEVEL ).reduce( ( supportedVersion, [ emoji, version ] ) => {
+		if ( isEmojiSupported( emoji ) && version > supportedVersion ) {
+			supportedVersion = version;
+		}
+
+		return supportedVersion;
+	}, 0 );
```

Comment:
```suggestion
	return Object.entries( EMOJI_SUPPORT_LEVEL )
		.reduce( ( currentVersion, [ emoji, newVersion ] ) => {
			if ( isEmojiSupported( emoji ) && newVersion > currentVersion ) {
				return newVersion;
			}
	
			return currentVersion;
		}, 0 );
```

However, this condition could be much simpler if `EMOJI_SUPPORT_LEVEL` were sorted from the lowest to highest values.

Reply:
Fixed in [7343da3](https://github.com/ckeditor/ckeditor5/pull/17864/commits/7343da3b2734aaf3d93b6e1e8e455c9a2bf881df)